libkbfs/kbfs_ops_concur_test.go

```diff
@@ -1900,3 +1900,160 @@ func TestKBFSOpsCancelGetFavorites(t *testing.T) {
+	config1, _, ctx, _ := kbfsOpsConcurInit(t, userName1, userName2)
+	//defer kbfsConcurTestShutdown(t, config1, ctx, cancel)
+
+	config2 := ConfigAsUser(config1, userName2)
+	//defer CheckConfigAndShutdown(t, config2)
+
+	name := userName1.String() + "," + userName2.String()
+
+	rootNode2 := GetRootNodeOrBust(ctx, t, config2, name, false)
+	kbfsOps2 := config2.KBFSOps()
+	ops2 := getOps(config2, rootNode2.GetFolderBranch().Tlf)
+	doStall := make(chan struct{}, 1)
+	unstall := make(chan struct{})
+	blocked := make(chan struct{})
+	paths := make(chan struct{})
+	snc := &stallingNodeCache{
+		NodeCache: ops2.nodeCache,
+		doStall:   doStall,
+		unstall:   unstall,
+		blocked:   blocked,
+		paths:     paths,
+	}
+	ops2.nodeCache = snc
+	ops2.blocks.nodeCache = snc
+
+	// u1 creates a file.
+	rootNode1 := GetRootNodeOrBust(ctx, t, config1, name, false)
+	kbfsOps1 := config1.KBFSOps()
+	fileNodeA1, _, err := kbfsOps1.CreateFile(
+		ctx, rootNode1, "a", false, NoExcl)
+	if err != nil {
+		t.Fatalf("Couldn't create file: %v", err)
+	}
+
+	// u2 syncs and then disables updates.
+	if err := kbfsOps2.SyncFromServerForTesting(
+		ctx, rootNode2.GetFolderBranch()); err != nil {
+		t.Fatal("Couldn't sync user 2 from server")
+	}
+	_, err = DisableUpdatesForTesting(config2, rootNode2.GetFolderBranch())
+	if err != nil {
+		t.Fatalf("Couldn't disable updates: %v", err)
+	}
+
+	// u2 writes to the file.
+	data := []byte{1, 2, 3}
+	err = kbfsOps1.Write(ctx, fileNodeA1, data, 0)
+	if err != nil {
+		t.Errorf("Couldn't write file: %v", err)
+	}
+	if err := kbfsOps1.Sync(ctx, fileNodeA1); err != nil {
+		t.Fatalf("Couldn't finish sync: %v", err)
+	}
+
+	// u2 tries to lookup the file, which will block until we drain
+	// the paths channel.
+	var wg sync.WaitGroup
+	wg.Add(1)
+	var fileNodeA2 Node
+	go func() {
+		defer wg.Done()
+		var err error
+		fileNodeA2, _, err = kbfsOps2.Lookup(ctx, rootNode2, "a")
+		if err != nil {
+			t.Errorf("Couldn't lookup a: %v", err)
+		}
+	}()
+	// Wait for the lookup to block.
+	select {
+	case <-blocked:
+	case <-ctx.Done():
+		t.Fatal("Timeout while waiting for lookup to block")
+	}
+
+	// u2 starts to sync but the sync is stalled while holding the
+	// block lock.
+	doStall <- struct{}{}
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		if err := kbfsOps2.SyncFromServerForTesting(
+			ctx, rootNode2.GetFolderBranch()); err != nil {
+			t.Errorf("Couldn't sync user 2 from server: %v", err)
+		}
+	}()
+
+	// Unblock the lookup.
+	select {
+	case <-paths:
+	case <-ctx.Done():
+		t.Fatal("Timeout while waiting for paths")
+	}
+
+	// Wait for the sync to block.  NOTE: To repro KBFS-1717, this
+	// call needs to go before we unblock the paths lookup.  However,
+	// with the fix for KBFS-1717, the test will hang if we do that
+	// since the Lookup holds blockLock while it gets the path.  So as
+	// is, this isn't a direct repro but it's still a test worth
+	// having around.
+	select {
+	case <-blocked:
```

Comment:
You can replace this with `case unstall <- struct{}{}` right? (see above comment)